src/serializer/ResidualHeapVisitor.js

```diff
@@ -581,7 +581,32 @@ export class ResidualHeapVisitor {
           if (object.intrinsicName === "global") continue; // Avoid double-counting
           this.visitObjectProperty(binding);
         }
-        // TODO #990: Fix additional functions handing of ModifiedBindings
+        // Handing of ModifiedBindings
+        for (let additionalBinding of modifiedBindings.keys()) {
+          let modifiedBinding: Binding = ((additionalBinding: any): Binding);
+          let residualBinding;
+          if (modifiedBinding.isGlobal) {
+            residualBinding = this.globalBindings.get(modifiedBinding.name);
+          } else {
+            let containingEnv = modifiedBinding.environment;
+            invariant(containingEnv instanceof DeclarativeEnvironmentRecord);
+            let bindMap = this.declarativeEnvironmentRecordsBindings.get(containingEnv);
+            if (bindMap) residualBinding = bindMap.get(modifiedBinding.name);
+          }
+          // Only visit it if there is already a binding
+          if (residualBinding && modifiedBinding.value !== residualBinding.value) {
```

Comment:
The condition `modifiedBinding.value !== residualBinding.value` is a bit dubious, I can see it's validity under the assumption that no other additional function modifies this binding value.